# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: d0b311007c03
# Full Hash: d0b311007c033e83824f5f6996a70ab9e870f31f
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Get RTCP BYE and RTP timeout handling working again (from Bug 1595479) r=mjf,dminor
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106145
# ==============================================================================

diff -r daa9eac709dc -r d0b311007c03 dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Fri Feb 19 16:20:38 2021 +0000
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Fri Feb 19 15:56:41 2021 -0600
@@ -64,6 +64,7 @@
 class WebrtcVideoConduit
     : public VideoSessionConduit,
       public webrtc::Transport,
+      public webrtc::RtcpEventObserver,
       public rtc::VideoSinkInterface<webrtc::VideoFrame>,
       public rtc::VideoSourceInterface<webrtc::VideoFrame> {
  public:
@@ -256,6 +257,12 @@
 
   void CollectTelemetryData() override;
 
+  void OnRtcpBye() override;
+
+  void OnRtcpTimeout() override;
+
+  void SetRtcpEventObserver(mozilla::RtcpEventObserver* observer) override;
+
  private:
   // Don't allow copying/assigning.
   WebrtcVideoConduit(const WebrtcVideoConduit&) = delete;
@@ -458,6 +465,9 @@
   // Tracking the attributes of received frames over time
   // Protected by mTransportMonitor
   dom::RTCVideoFrameHistoryInternal mReceivedFrameHistory;
+
+  // Accessed only on main thread.
+  mozilla::RtcpEventObserver* mRtcpEventObserver = nullptr;
 };
 }  // namespace mozilla
 